src/lib/queries/setPassword.ts

```diff
@@ -0,0 +1,66 @@
+import { knexController } from '../../database/database';
+import { log } from '../utils/log';
+import { redisClient } from '../auth/redis';
+import { hashPassword, passwordIsValid } from '../auth/auth';
+import { updateUser } from './user';
+import { sendEmail } from '../utils/sendEmail';
+import { SetPasswordBody } from '../../types/auth.types';
+
+const APPLICATION_URI: string | undefined = process.env.APPLICATION_URI;
+
+if (APPLICATION_URI === undefined) {
+  throw new Error('setPassword function has no application uri!');
+}
+
+export const handlePasswordSetRequest = async (
+  body: SetPasswordBody
+): Promise<void> => {
+  const userInfo: { id: string; email: string } | undefined =
+    await knexController('user')
```

Comment:
Voiko Knexille syöttää tyypitystä jotenkin esim näin


```suggestion
    await knexController<{id: string; email:string}>('user')
```

?

Reply:
emt

Reply:
Onhan toi vähän hölmö